src/net/posix-stack.cc

```diff
@@ -825,17 +847,23 @@ future<> posix_datagram_channel::send(const socket_address& dst, packet p) {
 
 udp_channel
 posix_network_stack::make_udp_channel(const socket_address& addr) {
-    return udp_channel(std::make_unique<posix_datagram_channel>(addr));
+    if (!addr.is_unspecified()) {
+        return make_bound_datagram_channel(addr);
+    } else {
+        // Preserve the default behavior of make_udp_channel({}) which is to
+        // create an unbound channel that can be used to send UDP datagrams.
+        return make_unbound_datagram_channel(AF_INET);
+    }
 }
 
```

Comment:
You could put this patch first, then the previous patch doesn't have the not-so-nice throw unimplemented.

Reply:
I agree that the throw is not so nice:/ I would still prefer to keep this order because introducing the interface first makes it easier to understand the motivation behind implementation.